linera-views/src/backends/rocks_db.rs

```diff
@@ -89,89 +90,55 @@ impl RocksDbStoreInternal {
         options.create_if_missing(true);
         let db = DB::open(&options, path)?;
         let root_key = root_key.to_vec();
+        let metrics = Handle::current().metrics();
+        let spawn_blocking = metrics.num_workers() == 1;
         Ok(RocksDbStoreInternal {
             db: Arc::new(db),
             _path_with_guard: path_with_guard,
             root_key,
             max_stream_queries,
             cache_size,
+            spawn_blocking,
         })
     }
-}
-
-impl WithError for RocksDbStoreInternal {
-    type Error = RocksDbStoreError;
-}
-
-impl ReadableKeyValueStore for RocksDbStoreInternal {
-    const MAX_KEY_SIZE: usize = MAX_KEY_SIZE;
-    type Keys = Vec<Vec<u8>>;
-    type KeyValues = Vec<(Vec<u8>, Vec<u8>)>;
-
-    fn max_stream_queries(&self) -> usize {
-        self.max_stream_queries
-    }
-
-    async fn read_value_bytes(&self, key: &[u8]) -> Result<Option<Vec<u8>>, RocksDbStoreError> {
-        ensure!(key.len() <= MAX_KEY_SIZE, RocksDbStoreError::KeyTooLong);
-        let client = self.clone();
-        let mut full_key = self.root_key.to_vec();
-        full_key.extend(key);
-        Ok(tokio::task::spawn_blocking(move || client.db.get(&full_key)).await??)
-    }
 
-    async fn contains_key(&self, key: &[u8]) -> Result<bool, RocksDbStoreError> {
-        ensure!(key.len() <= MAX_KEY_SIZE, RocksDbStoreError::KeyTooLong);
-        let client = self.clone();
-        let mut full_key = self.root_key.to_vec();
-        full_key.extend(key);
-        let key_may_exist =
-            tokio::task::spawn_blocking(move || client.db.key_may_exist(full_key)).await?;
+    fn inner_contains_key(&self, full_key: Vec<u8>) -> Result<bool, RocksDbStoreError> {
```

Comment:
I believe `contains_key_internal` is how we usually write it.

Reply:
Ok, renamed.